code/game/machinery/telecomms/telecomunications.dm

```diff
@@ -153,13 +153,12 @@ GLOBAL_LIST_EMPTY(telecomms_list)
 	else
 		on = FALSE
 		soundloop.stop()
+	if(old_on != on)
+		update_appearance()
 
 /obj/machinery/telecomms/process(delta_time)
 	update_power()
 
-	// Update the icon
```

Comment:
the appearance also relies on panel_open

Reply:
good point, implemented a setter